### ISSUE TITLE
No error messages when extending Enum that has variants

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#1696]
**Describe the bug**
Extending en `enum.Enum` that has defined variants produces no error messages:

**To Reproduce**

The following code produces no errors in pyrights's extension in VScode:

```python
import enum 

# pyright: strict

class A(enum.Enum):
    X = 123
    Y = 456

class B(A):
    Z = 789
```
But it is not  allowed during runtime.

**Expected behavior**
An error message when extending enum `A` via `class B(A)`

**VS Code extension or command-line**
VsCode Pyright extension v1.1.123. Also tested via `npx pyright` with version 1.1.126

**Comment by erictraut** [comment]
The Python documentation does indicate that an Enum class cannot be subclassed if the subclass defines new enum members (see https://docs.python.org/3/library/enum.html#restricted-enum-subclassing). However, this is getting into detailed semantics and behaviors of the Enum class implementation. In general, pyright does not hard code knowledge about behaviors of specific classes, so I'm reluctant to add this particular check. This goes beyond what a type checker would normally be expected to do.

I verified that mypy also does not include this check.
